src/Event.re

```diff
@@ -1,17 +1,17 @@
 /* Event.re */
 
-type cb('a) = 'a => unit;
+type unsubscribe = unit => unit;
 
-type t('a) = ref(list(cb('a)));
+type t('a) = ref(list('a => unit));
 
 let create = () => ref([]);
 
-let subscribe = (evt: t('a), f: cb('a)) => {
-  evt := List.append(evt^, [f]);
-  let unsubscribe = () => {
-    evt := List.filter(f => f !== f, evt^);
-  };
+let subscribe = (event, callback) => {
+  event := List.append(event^, [f]);
+
+  let unsubscribe = () =>
+    event := List.filter(cb => cb !== callback, event^);
   unsubscribe;
 };
 
-let dispatch = (evt: t('a), v: 'a) => List.iter(c => c(v), evt^);
+let dispatch = (event, message) => List.iter(cb => cb(message), event^);
```

Comment:
This module doesn't actually implement an "event" but rather an "event delegator". And the sloppy naming causes an issue here if you try to name the arguments properly, because what's passed in to be dispatched is what's usually considered the actual event. But since that name is taken I had to go with `message` here. Which isn't terrible, but also not all that accurate.